com.unity.uiwidgets/Runtime/painting/matrix_utils.cs

```diff
@@ -6,11 +6,59 @@
 
 namespace Unity.UIWidgets.painting {
     public static class MatrixUtils {
-        public static Offset getAsTranslation(this Matrix3 matrix3) {
-            return matrix3.isTranslate() ? new Offset(matrix3[Matrix3.kMTransX], matrix3[Matrix3.kMTransY]) : null;
+        public static float dot(this Vector3 left, Vector3 right) {
+            return left.x * right.x + left.y * right.y + left.z * right.z;
+        }
+        
+        public static Offset getAsTranslation(this Matrix4 values) {
```

Comment:
why only returns (x,y)? the translation of a matrix4 should be (x,y,z)?

Reply:
This is for matrix which only translate on x-y plane. The first if statement is used for check if it only make transform on x-y plane.